momento-sdk/src/main/java/momento/sdk/auth/StringCredentialProvider.java

```diff
@@ -21,8 +39,25 @@ public class StringCredentialProvider implements CredentialProvider {
    * @param authToken a Momento JWT.
    */
   public StringCredentialProvider(@Nonnull String authToken) {
-    this.authToken = authToken;
+    TokenAndEndpoints data;
```

Comment:
Nit: this and a few other variables can be final.

Reply:
The compiler won't let me make that final because it's potentially reassigned in the catch block. I did add `final` to the rest of the vars though.